src/python/pants/backend/project_info/filter_targets.py

```diff
@@ -52,6 +54,22 @@ def register_options(cls, register):
             metavar="[+-]regex1,regex2,...",
             help="Filter on targets with tags matching these regexes.",
         )
+        register(
+            "--type",
+            type=list,
+            metavar="[+-]type1,type2,...",
+            help="Filter on these target types, e.g. `resources` or `python_library`.",
+            removal_version="2.1.0.dev0",
+            removal_hint="Use `--target-type` instead of `--type`.",
+        )
+        register(
+            "--regex",
+            type=list,
+            metavar="[+-]regex1,regex2,...",
+            help="Filter on target addresses matching these regexes.",
+            removal_version="2.1.0.dev0",
+            removal_hint="Use `--address-regex` instead of `--regex`.",
+        )
```

Comment:
I'm not sure how to handle the v2 implementation not having `--target` and `--ancestor`. Should we have a deprecated option and then also error eagerly with a nice message when the value is passed?

Fwit, we could easily implement `--target`. I'm only confused the value compared to `--address-regex`. Meanwhile, `--ancestor` is harder to do.

Reply:
> Should we have a deprecated option and then also error eagerly with a nice message when the value is passed?

Yea, reasonable.